mmtrack/models/trackers/strongsort_tracker.py

```diff
@@ -0,0 +1,251 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import List, Tuple
+
+import numpy as np
+import torch
+from mmdet.structures.bbox import bbox_overlaps
+from mmengine.data import InstanceData
+from motmetrics.lap import linear_sum_assignment
+from torch import Tensor
+
+from mmtrack.registry import MODELS
+from mmtrack.structures import TrackDataSample
+from mmtrack.structures.bbox import bbox_xyxy_to_cxcyah
+from mmtrack.utils import OptConfigType, imrenormalize
+from .sort_tracker import SORTTracker
+
+
+def cosine_distance(x: Tensor, y: Tensor) -> np.ndarray:
+    """compute the cosine distance.
+
+    Args:
+        x (Tensor): embeddings with shape (N,C).
+        y (Tensor): embeddings with shape (M,C).
+
+    Returns:
+        ndarray: cosine distance with shape (N,M).
+    """
+    x = x.cpu().numpy()
+    y = y.cpu().numpy()
+    x = x / np.linalg.norm(x, axis=1, keepdims=True)
+    y = y / np.linalg.norm(y, axis=1, keepdims=True)
+    dists = 1. - np.dot(x, y.T)
+    return dists
+
+
+@MODELS.register_module()
+class StrongSORTTracker(SORTTracker):
+    """Tracker for StrongSORT.
+
+    Args:
+        obj_score_thr (float, optional): Threshold to filter the objects.
+            Defaults to 0.6.
+        reid (dict, optional): Configuration for the ReID model.
+            - num_samples (int, optional): Number of samples to calculate the
+                feature embeddings of a track. Default to None.
+            - image_scale (tuple, optional): Input scale of the ReID model.
+                Default to (256, 128).
+            - img_norm_cfg (dict, optional): Configuration to normalize the
+                input. Default to None.
+            - match_score_thr (float, optional): Similarity threshold for the
+                matching process. Default to 0.3.
+            - motion_weight (float, optional): the weight of the motion cost.
+                Defaults to 0.02.
+        match_iou_thr (float, optional): Threshold of the IoU matching process.
+            Defaults to 0.7.
+        num_tentatives (int, optional): Number of continuous frames to confirm
+            a track. Defaults to 2.
+    """
+
+    def __init__(self,
+                 obj_score_thr: float = 0.6,
+                 reid: dict = dict(
+                     num_samples=None,
+                     img_scale=(256, 128),
+                     img_norm_cfg=None,
+                     match_score_thr=0.3,
+                     motion_weight=0.02),
+                 match_iou_thr: float = 0.7,
+                 num_tentatives: int = 2,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.obj_score_thr = obj_score_thr
+        self.reid = reid
+        self.match_iou_thr = match_iou_thr
+        self.num_tentatives = num_tentatives
```

Comment:
can we not rewrite `__init__` function?

Reply:
because some default values are changed in strongsort_tracker, so maybe I can leave the  `__init__` here for clarity and delete the four `self.xxx = xxx` here?